src/Pmi/Datastore/Entity.php

```diff
@@ -1,229 +1,50 @@
-                }
-                $value = $arguments[0];
-                $value = self::convertType($field, $value);
-                $this->data[$field] = $value;
-                return $this;
-            } elseif ($action == 'get') {
-                if (isset($this->data[$field])) {
-                    return $this->data[$field];
-                } else {
-                    return null;
-                }
-            }
-        }
-        throw new Exception("Call to undefined method {$name} for class " . get_class());
+        $datastoreClient = new DatastoreClientHelper();
+        return $datastoreClient->fetchAll(static::getKind());
     }
 
-    private static function convertType($field, $value)
+    public static function fetchOneById($id)
     {
-        if (!self::fieldExists($field)) {
-            return $value;
-        }
-        $fields = self::getSchema()->getProperties();
-        if (empty($fields[$field]['type'])) {
-            return $value;
-        }
-        switch ($fields[$field]['type']) {
-            case \GDS\Schema::PROPERTY_DATETIME:
-                if (is_string($value)) {
-                    $value = new \DateTime($value);
-                }
-                break;
-        }
-        return $value;
+        $datastoreClient = new DatastoreClientHelper();
+        return $datastoreClient->fetchById(static::getKind(), $id);
     }
 
-    private static function fieldExists($field)
+    public function setData($data)
     {
-        $fields = self::getSchema()->getProperties();
-        return array_key_exists($field, $fields);
+        $this->data = $data;
     }
 
     public function save()
     {
-        $datastore = new Datastore();
-        $keyId = $datastore->insert(self::getSchema(), $this->data, $this->keyName, $this->keyId);
-        $this->setKeyId($keyId);
-        return $this;
-    }
-
-    public function delete()
-    {
-        if ($this->gdsEntity) {
-            $repository = new \GDS\Store(self::getSchema());
-            $repository->delete($this->gdsEntity);
-        }
-    }
-
-    public function loadFromArray(array $data)
-    {
-        foreach ($data as $field => $value) {
-            if (self::fieldExists($field)) {
-                $value = self::convertType($field, $value);
-                $this->data[$field] = $value;
-            }
-        }
-    }
-
-    private function loadFromGdsEntity($gdsEntity)
-    {
-        $this->loadFromArray($gdsEntity->getData());
-        if ($gdsEntity->getKeyId()) {
-            $this->setKeyId($gdsEntity->getKeyId());
-        }
-        if ($gdsEntity->getKeyName()) {
-            $this->setKeyName($gdsEntity->getKeyName());
-        }
+        $datastoreClient = new DatastoreClientHelper();
+        return $datastoreClient->insert(static::getKind(), $this->data);
     }
 
-    private static function buildSqlAndParameters(array $criteria)
+    public function update($id)
```

Comment:
For both the update and delete functions, you shouldn't have to pass in an ID, since the entity already represents the correct entity. The internal properties of this object should store the necessary information.